lib/functions/since.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class Since
+  def self.calculate_units
+    proc { |date, unit| define_case(date, unit) }
+  end
+
+  def self.define_case(date, unit)
+    case unit
+    when 'day'
+      days_from(date)
+    when 'month'
+      months_from(date)
+    when 'year'
+      years_from(date)
+    end
```

Comment:
consider using https://makandracards.com/makandra/1469-distance-of-time-in-what-you-like-days-months-years

Reply:
Thank you for the advise. This solution is really magic and short.